plugins/inputs/timex/timex.go

```diff
@@ -0,0 +1,101 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux
+
+package timex
+
+import (
+	_ "embed"
+	"fmt"
+	"time"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Timex struct {
+	Log telegraf.Logger `toml:"-"`
+}
+
+func (*Timex) SampleConfig() string {
+	return sampleConfig
+}
+
+func (*Timex) Gather(acc telegraf.Accumulator) error {
+	var timex unix.Timex
+	status, err := unix.Adjtimex(&timex)
+	if err != nil {
+		return fmt.Errorf("failed to get time adjtimex stats: %w", err)
+	}
+
+	// Check the return status for clock state
+	// https://github.com/torvalds/linux/blob/master/include/uapi/linux/timex.h
+	synced := status != unix.TIME_ERROR
+
+	// https://man7.org/linux/man-pages/man2/adjtimex.2.html
+	// Notes for frequency adjustment ppm
+	ppm16 := float64(65536 * time.Second / time.Microsecond)
+
+	// https://man7.org/linux/man-pages/man2/adjtimex.2.html
+	// validate the status to determine if the time is in nanoseconds or microseconds
+	// STA_NANO (0x2000): time is in nanoseconds
+	// STA_MICRO (0x4000): time is in microseconds
+	multiplier := int64(1000)
+	if (timex.Status & unix.STA_NANO) != 0 {
+		multiplier = int64(1)
+	}
+
+	statusOutput := ""
+	switch status {
+	case unix.TIME_OK:
+		statusOutput = "ok"
+	case unix.TIME_INS:
+		statusOutput = "insert"
+	case unix.TIME_DEL:
+		statusOutput = "delete"
+	case unix.TIME_OOP:
+		statusOutput = "progress"
+	case unix.TIME_WAIT:
+		statusOutput = "wait"
+	case unix.TIME_ERROR:
+		statusOutput = "error"
+	}
+
+	tags := map[string]string{
+		"status": statusOutput,
+	}
+
+	fields := map[string]interface{}{
+		"offset_ns":                    int64(timex.Offset) * multiplier, //nolint:unconvert // Conversion needed for some architectures
+		"frequency":                    timex.Freq,
```

Comment:
Hmm shouldn't this be divided by `ppm16`? According to the documentation the frequency is in PPM, isn't it? I would expect the frequency to be in Hertz...